include/iterlib/FutureIterator.h

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) 2016, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+#pragma once
+
+#include "iterlib/Iterator.h"
+
+namespace iterlib {
+
+/**
+ * Provides Iterator interface for arbitrary future which returns
+ * std::vector<Item> result.
+ * Particularly useful for data source APIs which do not have iterator interface
+ * and in tests.
+ */
+template <typename T = Item> class FutureIterator : public Iterator {
+ public:
+  explicit FutureIterator(folly::Future<std::vector<T>>&& f,
```

Comment:
The rocksdb iterator in #9 is such a lazy iterator. Sure, we can look into some lazy iterators for testing purposes (or just use rocksdb iterator for now).